### PR TITLE
remove beta flag check for v1beta1 object param,results and array result

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
# Changes

This commit removes the beta feature flag check for v1beta1 object param,results and array result. Beta features are considered to be 'stable' for v1beta1 CRDs, so we shouldn't check them.

Related TEP for more context: https://github.com/tektoncd/community/blob/main/teps/0033-tekton-feature-gates.md#new-api-fields

/kind bug

Signed-off-by: Yongxuan Zhang yongxuanzhang@google.com

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Remove beta feature flag check for v1beta1 object param, results and array result. Object param, results and array result will be enabled if the  enable-api-fields feature flag is not alpha for v1beta1 CRDs (e.g. Tasks and Pipelines)
```

### CODE REVIEW

**Comment by lbernick** [comment]
@Yongxuanzhang i think there are still some tests sticking around that use "alpha" for object params unnecessarily, e.g. https://github.com/tektoncd/pipeline/blob/ca608f24e722807529a649167e46ecda67df4129/pkg/apis/pipeline/v1beta1/taskrun_validation_test.go#L111. Can you look into this?

**Review by JeromeJu — APPROVED** [review]
Thanks @Yongxuanzhang , happy to lgtm after rebased

**Comment by Yongxuanzhang** [comment]
/hold
let me take a look

**Pull request merged**
